upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -17,19 +17,56 @@ limitations under the License.
 package vspheretasks
 
 import (
+	"bytes"
+	"fmt"
 	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+	"io/ioutil"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/model"
 	"k8s.io/kops/upup/pkg/fi"
 	"k8s.io/kops/upup/pkg/fi/cloudup/vsphere"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
 // AttachISO represents the cloud-init ISO file attached to a VMware VM
 //go:generate fitask -type=AttachISO
 type AttachISO struct {
-	Name *string
-	VM   *VirtualMachine
```

Comment:
VirtualMachine is a task. I wanted to add more members to AttachISO. The way the reflection based task dependency in works is that it thinks that every file in the task is a dependency. For example, when I added InstanceGroup and BootstratScript fields in AttachISO task, the reflection based task dependency thinks that InstanceGroup and BootstratScript are tasks and fails complaining that they are not tasks.

Basically if the tasks structs are simple, then the reflection based task dependency management is useful but in our case it is not simple. So, I had to explicitly call out the task dependency which is also fine.